manager/manager.py

```diff
@@ -0,0 +1,1105 @@
+import copy
```

Comment:
Nitpicking, but `isort` is very useful to organise imports by type. Maybe we should adopt the [brainreg pre-commit config](https://github.com/brainglobe/brainreg/blob/master/.pre-commit-config.yaml) to autoformat team projects?

Reply:
Thanks good idea I have added as pre-commit, I did run isort on it but maybe made changes following - thus good to have pre-commit!